hazelcast/src/main/java/com/hazelcast/logging/Logger.java

```diff
@@ -71,16 +71,40 @@ public static ILogger getLogger(Class clazz) {
      * @return the obtained logger.
      */
     public static ILogger getLogger(String name) {
-        // try the fast path first
         LoggerFactory existingFactory = loggerFactory;
-        if (existingFactory != null) {
+        if (loggerFactory != null) {
```

Comment:
We can get a NPE because of this change as `if` statement checks `loggerFactory`, but the internal block operates on `existingFactory`. The old code seems to be correct to me as it was reading a `volatile` fields into a local variable and then operating on it.